### PR TITLE
LOG-5621: Add pipeline validations

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Adds pipeline validation to ensure references exist 

### Links
* https://issues.redhat.com/browse/LOG-5621

cc @Clee2691 @vparfonov

### CODE REVIEW

File: internal/validations/observability/pipelines/validate.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		for _, field := range filter.PruneFilterSpec.NotIn {
+			if field == hostName {
+				found = true
+				break
+			}
+		}
+		if !found {
+			inListPrunes = true
+		}
+	}
```
Comment (vparfonov) [inline internal/validations/observability/pipelines/validate.go:97]:
```suggestion
	if filter.PruneFilterSpec.NotIn != nil {
		for _, field := range filter.PruneFilterSpec.NotIn {
			 if field == hostName {
			   inListPrunes = true
			   break
		  }
	  }
	}
```

Comment (jcantrill) [reply internal/validations/observability/pipelines/validate.go:97]: This is ported over from the previous tests and the recommended changes causes a test failure

Comment (vparfonov) [reply internal/validations/observability/pipelines/validate.go:97]: oh, yes, my bad, i missed ! in`if !found`

**Pull request merged**